presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -318,6 +319,11 @@ public QueryInfo createQuery(Session session, String query)
         queryMonitor.createdEvent(queryInfo);
 
         queryExecution.addStateChangeListener(newValue -> {
+            if (newValue.equals(FINISHED)) {
```

Comment:
This should be `isDone()` since we want to always notify, even if the query fails